### PR TITLE
Added Go embed demo (Go 1.16) [WIP] [DO NOT MERGE]

### PR DESCRIPTION
**Pull request by eth0net**
Added a demo to show how to embed assets using the new Go embed package.

Does require go.mod to be set to Go 1.16 so this should probably be held back for a while.

### CODE REVIEW

**Comment by raziel2244** [comment]
Ah yeah forgot those would need updating, updating those now.

**Comment by raziel2244** [comment]
Okay so still not passing... think I'll just keep this aside for later. Once 1.16 is more established and Engo maybe gets a go.mod update for some reason besides this it can be looked into again.

For anyone looking to embed using Go embed, examples can be provided elsewhere worst case.

**Comment by raziel2244** [comment]
Reopened as WIP to keep it tracked here for the future.

**Comment by Noofbiz** [comment]
I fixed the traffic manager issues, so they should pass if they re-run. The only issue I see is I think engo's go mod version should stay 1.14. Even though 1.16 is required for the embedding, that's not a feature engo itself uses so it shouldn't need to increase the version for it.

File: go.mod [hunk 1]
```diff
@@ -31,4 +31,4 @@ require (
 	gopkg.in/yaml.v2 v2.2.8 // indirect
 )
 
-go 1.14
+go 1.16
```
Comment (Noofbiz) [inline go.mod:34]: This shouldn't need to be updated, only testing

**Review by Noofbiz — CHANGES_REQUESTED** [review]

**Comment by raziel2244** [comment]
Thanks, yeah I was thinking it's probably best to leave the go version for Engo as 1.14. Unfortunately, since the demo is within the module it fails to build without the update to go.mod.

**Comment by raziel2244** [comment]
Just looking to see if there is a way to set the -lang flag for build as that could be a way round it, maybe.

**Comment by raziel2244** [comment]
Can't see a way around it for now, unfortunately. The demo can live elsewhere in the meantime though, such as just in the README file.

**Comment by Noofbiz** [comment]
Alright, got the CI stuff working. Turns out the go.mod includes the demos, but it's not a hard/fast rule. I think 1.16 is mature enough at this point to move up to it if you want to merge this in. Looks good to me!

**Review by Noofbiz — APPROVED** [review]

**Comment by eth0net** [comment]
Ah well if you're happy to up the Go version then go ahead with the merge, I believe this is all ready!

**Pull request merged**